### PR TITLE
add type proxy for optional properties

### PR DESCRIPTION
**Pull request by bbergstrum**

### CODE REVIEW

File: src/combinator.ts [hunk 1]
```diff
@@ -107,6 +108,8 @@ export const or4P = <A, B, C, D>(
   return or2P(or2P(first, second), or2P(third, fourth));
 };
 
+export const optionalP = <T>(proxy: TypeProxy<T>) => or2P(undefinedP, proxy);
```
Comment (deanylev) [inline src/combinator.ts:111]: I think the variable name `proxy` should be `type` right? To match the other functions

Comment (bbergstrum) [reply src/combinator.ts]: yeah, updated.

**Review by andrewjpritchard — APPROVED** [review]

File: src/combinator.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { pureP, TypeProxy } from '.';
 import { ParseError } from './error';
+import { undefinedP } from './primitive';
```
Comment (andrewjpritchard) [inline src/combinator.ts:3]: You should increment the version number in package.json

Comment (bbergstrum) [reply src/combinator.ts:3]: done

Comment (dale-vendia) [reply src/combinator.ts:3]: nice

**Pull request merged**